src/goSignature.ts

```diff
@@ -94,12 +94,18 @@ export class GoSignatureHelpProvider implements SignatureHelpProvider {
 		for (let lineNr = position.line; lineNr >= 0 && maxLookupLines >= 0; lineNr-- , maxLookupLines--) {
 
 			const line = document.lineAt(lineNr);
+
+			// Stop processing if we're inside a comment
+			if (isPositionInComment(document, position)) {
+				return null;
+			}
+
 			// if its current line, get the text until the position given, otherwise get the full line.
 			const [currentLine, characterPosition] = lineNr === position.line
 				? [line.text.substring(0, position.character), position.character]
 				: [line.text, line.text.length - 1];
 
-			for (let char = characterPosition - 1; char >= 0; char--) {
+			for (let char = characterPosition; char >= 0; char--) {
```

Comment:
Sometime during the last year I introduced this regression with the belief this would prevent an issue in the future, and while even though the code path shouldn't have caused any issues I remained none the wiser :man_facepalming:

Sorry for that. This should fix it

Reply:
No problem @lggomez, thanks for the fix!